src/System.IO.Pipelines/ReadCursorOperations.cs

```diff
@@ -2,371 +2,94 @@
-                            {
-                                return -1;
-                            }
-
-                            continue;
-                        }
-
-                        if (block == end.Segment && index + byteIndex >= end.Index)
-                        {
-                            // Ensure iterator is left at limit position
-                            return -1;
-                        }
-
-                        result = new ReadCursor(block, index + byteIndex);
-                        return array.Array[array.Offset + index + byteIndex];
-                    }
-                    // Need unit tests to test Vector path
-#if !DEBUG
-                    }
-#endif
-                    fixed (byte* pCurrentFixed = array.Array)
-                    {
-                        var pCurrent = pCurrentFixed + array.Offset + index;
-
-                        var pEnd = block == end.Segment ? pCurrentFixed + array.Offset + end.Index : pCurrent + following;
-                        while (pCurrent < pEnd)
-                        {
-                            if (*pCurrent == byte0)
-                            {
-                                result = new ReadCursor(block, index);
-                                return byte0;
-                            }
-                            if (*pCurrent == byte1)
-                            {
-                                result = new ReadCursor(block, index);
-                                return byte1;
-                            }
-                            pCurrent++;
-                            index++;
-                        }
-                    }
-
-                    following = 0;
-                    break;
+                    result = new ReadCursor(segment, enumerator.CurrentSegmentStartIndex + index);
+                    return span[index];
                 }
             }
+
+            result = end;
+            return -1;
         }
 
-        public static unsafe int Seek(ReadCursor begin, ReadCursor end, out ReadCursor result, byte byte0, byte byte1, byte byte2)
+        public static int Seek(ReadCursor begin, ReadCursor end, out ReadCursor result, byte byte0, byte byte1, byte byte2)
         {
-            result = end;
-            var block = begin.Segment;
-            if (block == null)
+            var enumerator = new MemoryEnumerator(begin, end);
+            while (enumerator.MoveNext())
             {
-                return -1;
-            }
+                var span = enumerator.Current.Span;
+                var segment = enumerator.CurrentSegment;
 
-            var index = begin.Index;
-            var wasLastBlock = block.Next == null;
-            var following = block.End - index;
-            int byteIndex = int.MaxValue;
+                int index1 = span.IndexOfVectorized(byte0);
+                int index2 = span.IndexOfVectorized(byte1);
+                int index3 = span.IndexOfVectorized(byte2);
```

Comment:
Yes we know it's inefficient, the bug is here https://github.com/dotnet/corefxlab/issues/1232. Luckily, we don't call this anywhere in kestrel right now.